apps/aesophia/src/aeso_abi.erl

```diff
@@ -52,6 +54,8 @@ encode_function(_Contract, Function) ->
 
 %% TODO: Handle all sophia data types.
 get_type(I) when is_integer(I) -> int;
+get_type(none) -> nil;
+get_type({some, X}) -> {option, get_type(X)};
```

Comment:
I see `option`, `None` and `Some` first mentioned [at this point in the spec](https://github.com/aeternity/protocol/blob/cde08c291750614a5908973100657ccbd69bf92f/contracts/sophia.md#algebraic-data-types). Maybe they shall be defined before that?